File: 38주차/서민주/ContainerWithMostWater.swift
class Solution {
    func maxArea(_ height: [Int]) -> Int {
        var maxArea = 0
        var (i, j) = (0, height.count-1)
        
        while i < j {
            maxArea = max(maxArea, min(height[i], height[j]) * (j-i))
            if height[i] < height[j] { i += 1 } 
            else { j -= 1 }
        }

        return maxArea
    }
}

File: 37주차/안영진/ContainerWithMostWater.java
package me.hellozin.ps.leetcode.medium;

// https://leetcode.com/problems/container-with-most-water/

public class ContainerWithMostWater {

    public int maxArea(int[] height) {
        int left = 0;
        int right = height.length - 1;

        int max = 0;
        while (left < right) {
            int area = (right - left) * Math.min(height[left], height[right]);
            max = Math.max(area, max);
            if (height[left] < height[right]) {
                left++;
            } else {
                right--;
            }
        }
        return max;
    }

}
